### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by TomaszPartyka0990**
I secured app with Spring Security, implemented authentication from DB, made current login page, register form for adding users and validated it.
It looks horrible but views is my last step :)
On next step i want to implement exception handling and then i can focus on functionality.

### CODE REVIEW

**Comment by TomaszPartyka0990** [comment]
Added profiles and some error handling. Can I merge ? 
Want to move on with events adding and the rest of main functionality

**Comment by TomaszPartyka0990** [comment]
Please review the code

File: .gitignore [hunk 1]
```diff
@@ -28,3 +28,6 @@ hs_err_pid*
 *.iws
 *.iml
 *.ipr
+
+#Ignoring properties with DB config
+/src/main/resources/application-prod.properties
```
Comment (kchrusciel) [inline .gitignore:33]: You can add information about this in README.

Comment (kchrusciel) [reply .gitignore:33]: Can you add also information to README how to run it?

File: src/main/java/pl/sda/partyka/DataInitializer.java [hunk 2]
```diff
@@ -1,21 +1,63 @@
...
-        service.addRole(new Role("ORGANIZER"));
-        service.addRole(new Role("USER"));
+        if (runningProfile.equals(DEVELOP_PROFILE)) {
+            Role organizer = roleService.addRole(new Role("ORGANIZER"));
+            roleService.addRole(new Role("USER"));
+            UserCreateRequest userCreateRequest = new UserCreateRequest();
+            userCreateRequest.setLogin("asd@asd.pl");
+            userCreateRequest.setPassword("12345678");
+            userCreateRequest.setRepeatedPassword("12345678");
+            userCreateRequest.setDisplauName("ASD");
```
Comment (kchrusciel) [inline src/main/java/pl/sda/partyka/DataInitializer.java:42]: Typo

File: pom.xml [hunk 3]
```diff
@@ -66,18 +65,28 @@
             <artifactId>spring-boot-starter-thymeleaf</artifactId>
             <version>2.3.4.RELEASE</version>
         </dependency>
-        <!--
+
+        <dependency>
```
Comment (kchrusciel) [inline pom.xml:69]: For UI part you can add DevTools also

File: src/main/java/pl/sda/partyka/controller/CommentController.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+@Controller
+@RequiredArgsConstructor
+public class CommentController {
+
+    private final CommentService commentService;
+    private final UserService userService;
+    private final EventService eventService;
+
+    @PostMapping("/addComment")
```
Comment (kchrusciel) [inline src/main/java/pl/sda/partyka/controller/CommentController.java:27]: Optional: use `/comments` mapping

File: src/main/java/pl/sda/partyka/controller/CommentController.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    public String addComment(@ModelAttribute (value = "commentCreateRequest") @Valid CommentCreateRequest commentCreateRequest,
+                             Errors validationErrors, RedirectAttributes redirectAttributes, Principal principal){
+        if (validationErrors.hasErrors()){
+            ControllersMethods.getErrorsAddThemToAttributes(validationErrors, redirectAttributes);
+            return "redirect:/events/" + commentCreateRequest.getEventId();
+        }
+        User author = userService.getUserByLogin(principal.getName());
+        author.getCreatedComments().forEach(comment -> System.out.println(comment.getContent()));
```
Comment (kchrusciel) [inline src/main/java/pl/sda/partyka/controller/CommentController.java:35]: You can use logger -> check how to logging in the Spring Boot

File: src/main/java/pl/sda/partyka/controller/ControllersMethods.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+import pl.sda.partyka.error.ErrorInfo;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class ControllersMethods {
+
+    static void getErrorsAddThemToAttributes(Errors validationErrors, RedirectAttributes redirectAttributes){
```
Comment (kchrusciel) [inline src/main/java/pl/sda/partyka/controller/ControllersMethods.java:19]: This method can be extraced to class ErrorCreator or something else (we can move it also to error package). We should avoid God's classes.

File: src/main/java/pl/sda/partyka/controller/EventContoller.java [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+import javax.validation.Valid;
+import java.security.Principal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+@Controller
+@RequiredArgsConstructor
+public class EventContoller {
```
Comment (kchrusciel) [inline src/main/java/pl/sda/partyka/controller/EventContoller.java:31]: Typo in class name, should be Controller

File: src/main/java/pl/sda/partyka/controller/EventContoller.java [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+    private final EventService eventService;
+    private final UserService userService;
+    private final CommentService commentService;
+
+    @Secured("ROLE_ORGANIZER")
+    @GetMapping("/addEvent")
+    public String showAddingEventForm(Model model, @ModelAttribute(value = "errors") ArrayList<ErrorInfo> errorsCaught){
```
Comment (kchrusciel) [inline src/main/java/pl/sda/partyka/controller/EventContoller.java:39]: Optional: use List instead of ArrayList

**Review by kchrusciel — APPROVED** [review]
Looks good to me, you should think about tests!

**Pull request merged**